[Ong Ying Gao] iP

ong-yinggao98 opened the pull request.

Commit:
Add Gradle support

ong-yinggao98 committed:
Changed greeting message

ong-yinggao98 committed:
Added bye command

ong-yinggao98 committed:
Added ability to record input strings from user as a task

ong-yinggao98 committed:
Added list command to view recorded tasks

ong-yinggao98 committed:
Added Task class

ong-yinggao98 committed:
Reorganised tasks into their respective Task objects

ong-yinggao98 committed:
Added ability to complete tasks

ong-yinggao98 committed:
Removed unnecessary else clause

ong-yinggao98 committed:
Silly me, re-added ability to add tasks

ong-yinggao98 committed:
Added deadlines, events and todos

ong-yinggao98 committed:
Added ability to add different types of tasks

ong-yinggao98 committed:
Reorganised commands into their own methods

ong-yinggao98 committed:
Fixed grammatical errors

ong-yinggao98 committed:
Added automated testing

ong-yinggao98 committed:
Updated runtest.bat

ong-yinggao98 committed:
Added exception classes and static factory methods for error handling

ong-yinggao98 committed:
Updated test cases, fixed bug at Event.java

ong-yinggao98 committed:
Added delete command

ong-yinggao98 committed:
Refactored exceptions as their own package

ong-yinggao98 committed:
Refactored tasks into their own package

ong-yinggao98 committed:
imported newly created packages

ong-yinggao98 committed:
Add new packages into runtest.bat and runtest.sh

ong-yinggao98 committed:
Add save functionality

ong-yinggao98 committed:
Set fields in Deadline class to private

ong-yinggao98 committed:
Add date recognition

ong-yinggao98 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/task/Deadline.java
#	src/main/java/task/Event.java
#	src/main/java/task/Task.java

ong-yinggao98 committed:
Add Ui, Parser, Storage, TaskList classes

ong-yinggao98 committed:
Refactor classes into duke.main, duke.exception and duke.task

ong-yinggao98 committed:
Change Parser.parse() message on adding a new task

ong-yinggao98 committed:
Fix incorrect package names

ong-yinggao98 committed:
Add JUnit tests

Test cases test public and package-private methods in duke.task package as part of A-JUnit requirements.

ong-yinggao98 committed:
Resolve errors when printing or writing empty TaskLists

ong-yinggao98 committed:
Add JavaDoc comments to all non-trivial non-private methods

ong-yinggao98 committed:
Add support for find operation

Methods have been added to Task, Parser and TaskList classes to facilitate searching for Tasks that contain a given keyword.

ong-yinggao98 committed:
Amend code style

Braces have been added to single-line if clauses in compliance with the coding standards.

ong-yinggao98 committed:
Merge branch 'branch-A-JavaDoc'

ong-yinggao98 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/java/duke/task/Event.java

ong-yinggao98 committed:
Rename factory methods for file reading

ong-yinggao98 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/java/duke/main/TaskList.java

ong-yinggao98 committed:
Fix bug in Parser.find()

ong-yinggao98 committed:
Merge branch 'add-gradle-support' of https://github.com/StopTakingAllTheNames/ip into branch-A-Gradle

ong-yinggao98 committed:
Add checkstyle to detect coding violations

ong-yinggao98 committed:
Fix missing main module

ong-yinggao98 committed:
Merge branch 'branch-A-Gradle'

ong-yinggao98 committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat
#	src/test/java/duke/task/DeadlineTest.java

ong-yinggao98 committed:
Resolve coding standard violation

LeeMingDe commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,28 @@
+package duke.main;
+
+public class Duke {
```
Should there be java docs for this class?

LeeMingDe reviewed (commented):
Hi, your code is clean and tidy. I think that some classes are missing java docs, so it would be good to add them in. Otherwise, keep up the good work!

kormingsoon commented on src/main/java/duke/exception/EmptyTaskException.java:
```diff
@@ -0,0 +1,14 @@
+
+public class EmptyTaskException extends Exception {
+    private String taskType;
+
```
Think you can javadoc here if you want LOL

ong-yinggao98 replied:
Damn, I can't believe I had missed out the point where all classes need JavaDoc headers!

kormingsoon commented on src/main/java/duke/exception/InvalidDateException.java:
```diff
@@ -0,0 +1,9 @@
+    @Override
+    public String toString() {
+        return "Invalid date format detected!\n"
+                + "Accepted formats: 'dd/MM/yyyy' or 'dd-MM-yyyy";
```
Like the formatting

kormingsoon commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,28 @@
+     * @param args
+     */
+    public static void main(String[] args) {
+        new Duke("data/tasks.txt").run();
```
Interesting way of running Duke haha

kormingsoon commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,118 @@
+
+public class Parser {
+    private final TaskList taskList;
+    private StringBuilder reply = new StringBuilder();
```
Ooo good use of StringBuilder

kormingsoon commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,118 @@
+     * @param input User command in console.
+     * @return Output message upon execution of command.
+     */
+    String parse(String input) {
```
Access modifier can be added here

VaishakAnand replied:
Agreed ^

kormingsoon commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,118 @@
+    String parse(String input) {
+        try {
+            if (input.equals("list")) {
+                list();
```
Would suggest to use this.list(), was abit confused where the method came from when I read it initially

kormingsoon commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,118 @@
+        reply = new StringBuilder();
+        return output.toString();
+    }
+
```
I would recommend you to create the Command class to better abstract commands such as ListCommand DeleteCommand etc, will help in shortening Parser.java.

kormingsoon commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,118 @@
+            try {
+                int index = Integer.parseInt(input.substring(7)) - 1;
+                Task deleted = taskList.delete(index);
+                reply.append("Noted, I have removed the below task:")
```
I would recommend concatenating the string or using your StringBuilder to construct your input as a whole instead of appending so many times

kormingsoon commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,54 @@
+     *
+     * @return TaskList representing all the tasks in the saved file (if any).
+     */
+    TaskList getTasksFromFile() {
```
Access modifier

kormingsoon commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -0,0 +1,190 @@
+     *
+     * @return String representation of a saved TaskList.
+     */
+    String write() {
```
access modifier

kormingsoon commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,41 @@
+import java.util.Scanner;
+
+public class Ui {
+    public static final String LINE = "____________________";
```
great use of static variables here

kormingsoon commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,41 @@
+     * input is parsed into a Parser object and interpreted, after which
+     * its output is displayed back to the user.
+     */
+    void run() {
```
access modifier

kormingsoon commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,92 @@
+    @Override
+    public String print() {
+        return "D | "
+                + (isDone ? 1 : 0) + " | "
```
think you can abstract a separate method here

kormingsoon commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,93 @@
+        String dateTime = date.format(DateFormat.FORMAT5.toDateFormat());
+        return "[E]" + super.toString() + " (at: " + dateTime + ")";
+    }
+
```
feels like there are some repetition you can abstract to the super class.

VaishakAnand replied:
^ I disagree, I think all the needed abstraction is already done. I might be wrong tho

kormingsoon commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,33 @@
+            fail();
+        }
+    }
+
```
nice testing

kormingsoon reviewed (commented):
Great effort overall but I think you can abstract some common aspects between a few classes for better abstraction and readability. Can also be more consistent with your access modifiers.

ong-yinggao98 committed:
Add JavaFX GUI

ong-yinggao98 committed:
Merge branch 'branch-Level-10'

ong-yinggao98 committed:
Add JavaDoc comments to all public classes

VaishakAnand commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,121 @@
+     */
+    String parse(String input) {
+        try {
+            if (input.equals("list")) {
+                list();
+            } else if (input.startsWith("done")) {
+                done(input);
+            } else if (input.startsWith("delete")) {
+                delete(input);
+            } else if (input.startsWith("find")) {
+                find(input);
+            } else {
+                add(input);
+            }
```
This usage of if - else works for sure, but perhaps the use of switch - cases would be more appropriate in this case.
Minor issue tho xD

ong-yinggao98 replied:
Thanks for the suggestion! I forgot I could simply extract the first word with a scanner instead of using `String.startsWith()`. Not that it matters now that we're using GUI, but at least I've found my own way of doing so.

VaishakAnand commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,97 @@
+        }
+
+        LocalDate dateTime = null;
+        for (DateFormat format : DateFormat.values()) {
+            try {
+                dateTime = LocalDate.parse(taskInfo[1], format.toDateFormat());
+            } catch (DateTimeParseException ignored) {
+                continue;
+            }
+        }
```
Great use of enhanced for-loop and enum to allow for multiple date formats input by the user 👍 
Did not think of this at all :)

VaishakAnand commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public abstract class Task {
+    protected final String description;
+    protected boolean isDone;
```
This access modifier can be set to private, since there's a setter method and a getter method for the correct symbols already implemented.

ong-yinggao98 replied:
Good catch!

VaishakAnand reviewed (commented):
In general, I could not find any issue with naming, and you had minimal coding standards violations overall!

I only had to make minor suggestions, and other issues have already been pointed out by the other reviewer.

Great job :)

ong-yinggao98 committed:
Set Parser and Ui method to public

Affected methods:
* `Parser()`
* `Parser.parse()`
* `Ui()`

ong-yinggao98 committed:
Rename methods to be more specific

Affected methods:
* `Task.print()` to `Task.toDataString()`
* `TaskList.write()` to `TaskList.toSaveFormat()`
* `Storage.write()` to `Storage.writeToFile()`

ong-yinggao98 committed:
Use switch cases in Parser class to directly operate on TaskList

When input is received, Parser extracts out the first word in the command to decide what to do with the remaining command details.

ong-yinggao98 committed:
Deprecate EmptyTaskException and Ui

ong-yinggao98 committed:
Improve abstraction for Task factory methods

Changed:
* Creation of dates for Deadlines and Events now handled by static DateFormat.getLocalDate()
* createFromFile() methods now throw exceptions if the saved data is in the wrong format.

ong-yinggao98 committed:
Resolve style violations

ong-yinggao98 committed:
Update Test classes to test changed static factory methods

ong-yinggao98 committed:
Update EventTest to test modified static factory method

ong-yinggao98 committed:
Add greeting message

Added:
# Greeting message upon initialisation of chatbot
Fixed:
# Text wrapping being lost when scroll bar appears

ong-yinggao98 committed:
Add assertions

ong-yinggao98 committed:
Improve code quality

ong-yinggao98 committed:
Remove unused code

ong-yinggao98 committed:
Add duke.command package to abstract commands

Added:
* AddDeadline, AddEvent, AddToDo, AddTaskCommand
* FindCommand
* DeleteCommand
* ByeCommand
* ListCommand
* FindCommand
* DoneCommand
* Command
Improved:
* Add more JavaDoc comments to improve style

ong-yinggao98 committed:
Merge branch 'master' into branch-A-Assertions

ong-yinggao98 committed:
Merge pull request #1 from StopTakingAllTheNames/branch-A-Assertions

Add assertions

ong-yinggao98 committed:
Merge branch 'master' into branch-A-CodeQuality

ong-yinggao98 committed:
Merge pull request #2 from StopTakingAllTheNames/branch-A-CodeQuality

Improve Code Quality

ong-yinggao98 committed:
Merge branch 'master' into branch-A-Assertions

 Add CI support

ong-yinggao98 committed:
Add support for "help" command

ong-yinggao98 committed:
Add detection of duplicates

New tasks can only be added iff a similar task does not exist in the tasklist prior.
Additionally, the creation of new deadlines and events have been fixed.

ong-yinggao98 committed:
Merge branch 'branch-C-Help'

ong-yinggao98 committed:
Merge branch 'branch-C-DetectDuplicate'

# Conflicts:
#	src/main/java/duke/main/Parser.java

ong-yinggao98 committed:
Improve styling on test classes

ong-yinggao98 committed:
Remove obsolete I/O redirection tests

ong-yinggao98 committed:
Remove I/O tests in CI workflow (+1 squashed commit)
Squashed commits:
[37f791b] Remove obsolete I/O redirection tests

ong-yinggao98 committed:
Merge commit '37f791bf3182c4112b9af6d46233a48ed42d5b04'

ong-yinggao98 committed:
Credit reused code and adapted solutions

ong-yinggao98 committed:
Add User Guide

Changes:
* Rename "bye" command to "save"
* Add title to application
* Add screenshot for website

ong-yinggao98 committed:
Fix incorrect import order in Parser.java

ong-yinggao98 committed:
Update archive name and version
